src/sidebar/components/hooks/test/use-element-should-close-test.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const { createElement } = require('preact');
+const { useRef } = require('preact/hooks');
+const propTypes = require('prop-types');
+
+const { act } = require('preact/test-utils');
+const { mount } = require('enzyme');
+
+const useElementShouldClose = require('../use-element-should-close');
+
+describe('hooks.useElementShouldClose', () => {
+  let handleClose;
+  let e;
+  const events = [
+    new Event('mousedown'),
+    new Event('click'),
+    ((e = new Event('keypress')), (e.key = 'Escape'), e),
+    new Event('focus'),
+  ];
+
+  // Create a fake component to mount in tests that uses the hook
+  function FakeComponent({ isOpen = true }) {
+    const myRef = useRef();
+    useElementShouldClose(myRef, isOpen, handleClose);
```

Comment:
I think this approach is fine, if we were to extend this even more, I could imagine we could make a component that bakes in this hook and then it becomes a wrapper around anything we want the have that hook behavior.

```
<CanBeClosed>
   <div>some stuff... </div>
</CanBeClosed>
```